crawl-ref/source/item-name.cc

```diff
@@ -1952,6 +1952,31 @@ bool item_type_known(const object_class_type base_type, const int sub_type)
     return you.type_ids[base_type][sub_type];
 }
 
+// Has every item a scroll can identify (i.e. all scrolls and potions) been
+// identified by "you"?
+static bool _is_everything_identified;
+
+void check_if_everything_is_identified(void)
+{
+    const object_class_type types[] = {OBJ_SCROLLS, OBJ_POTIONS};
```

Comment:
This function could return immediately if its condition (however implemented) is already marked as satisfied, as long as it has a default initialization.